sphinx/domains/cpp.py

```diff
@@ -2600,9 +2600,11 @@ def _stringify(self, transform: StringifyTransform) -> str:
             if self.volatile:
                 res.append(' ')
             res.append('const')
-        if self.const or self.volatile or len(self.attrs) > 0:
-            if self.next.require_space_after_declSpecs():
-                res.append(' ')
+        if (
+            (self.const or self.volatile or len(self.attrs) > 0) and
+            self.next.require_space_after_declSpecs()
+        ):
+            res.append(' ')
```

Comment:
Should not be changed, the original better conveys the semantics and is more readable.

Reply:
can you explain why you think the original better conveys the semantics?

to me they convey identical semantics, but with one less level of nesting.

not saying you're wrong, just wanting to hear your reasoning.

perhaps the meaning of this code is better conveyed by *naming* the boolean checks here with an assignment before the `if` statement, then by any particular arrangement of the statements?

Reply:
The outer condition is whether there is a local grammatical construct (CV-qualifiers and attributes) which means we should even consider emitting a space. The inner condition is whether a nested grammatical construct (the next declarator) is of a type which necessitates a space.
Stylistically the change means it takes up more lines and introduces parens, which I both find less readable as well for this case.